source/custom/xnetsocket_converters.h

```diff
@@ -282,6 +282,100 @@ inline SockOptDataInputConverter convert_from_grpc(const SockOptData& input)
   return SockOptDataInputConverter(input);
 }
 
+struct AddrInfoHintInputConverter {
+  AddrInfoHintInputConverter(const AddrInfoHint& input)
+  {
+    addr_info.ai_flags = input.flags();
+    addr_info.ai_family = input.family();
+    addr_info.ai_socktype = input.sock_type();
+    addr_info.ai_protocol = input.protocol();
+
+    // Other addr_info fields for the hint should be left unset (0 initialized).
+  }
+
+  operator nxaddrinfo*()
+  {
+    return &addr_info;
+  }
+
+  operator const nxaddrinfo*() const
+  {
+    return &addr_info;
+  }
+
+  nxaddrinfo addr_info{};
```

Comment:
I _think_ this is the proper way for the addr_info's fields to be 0-initialized.

Reply:
Yeah, nice one. I think that's 4.2 here: https://en.cppreference.com/w/cpp/language/value_initialization . It's hard to hunt down the various flavors of "braced initialization".

![image](https://user-images.githubusercontent.com/56981669/158883514-ddee5701-8e77-4515-91d6-9ac692b89ed3.png)

You can also play around with it in compiler explorer:

![image](https://user-images.githubusercontent.com/56981669/158883793-c3f6e163-c7a6-4397-aa26-e004d6bf9a02.png)

Interestingly it zeroes those 4 int32s with 2 Quad-word moves instead of 4 double-word moves.